### PR TITLE
Parser: support PURGE BINARY LOGS statement

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Adding `sqlparser` support for `PURGE BINARY LOGS TO|BEFORE` statement, in compliance with https://dev.mysql.com/doc/refman/8.0/en/purge-binary-logs.html

There is no implementation, and vitess does not respond to this query. I have a use case that will benefit from the completeness of supporting this query parser/ast -wise.

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Review by systay — APPROVED** [review]
LGTM

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -7570,6 +7584,7 @@ non_reserved_keyword:
 | AVG %prec FUNCTION_CALL_NON_KEYWORD
 | AVG_ROW_LENGTH
 | BEGIN
+| BEFORE
```
Comment (GuptaManan100) [inline go/vt/sqlparser/sql.y:7587]: `BEFORE` should come before `BEGIN` in alphabetic order

**Review by GuptaManan100 — APPROVED** [review]
LGTM otherwise

File: go/vt/sqlparser/sql.y (hunk 1 above)
Comment (shlomi-noach) [reply go/vt/sqlparser/sql.y]: fixed

**Pull request merged**